Automattic-Tracks-iOS/Model/Core Data/TracksContextManager.m

```diff
@@ -59,6 +74,64 @@ - (NSPersistentStoreCoordinator *)persistentStoreCoordinator {
     return _persistentStoreCoordinator;
 }
 
+- (NSURL *)storeURL {
+    return [[self storeContainerDirectoryURL] URLByAppendingPathComponent:@"Tracks.sqlite"];
+}
+
+- (NSURL *)storeContainerDirectoryURL {
+    if (self.sandboxed == YES) {
+        return [self applicationSupportURLForContainerApp];
+    } else {
+        return [self applicationDocumentsDirectory];
+    }
+}
+
+- (NSURL *)applicationSupportURLForContainerApp {
+    // The container app is the one owning the main bundle
+    return [self applicationSupportURLForAppWithBundleIdentifier:[[NSBundle mainBundle] bundleIdentifier]];
+}
+
+- (NSURL *)applicationSupportURLForAppWithBundleIdentifier:(NSString *)bundleIdentifier {
+    NSURL *folder = [[self applicationSupportURL] URLByAppendingPathComponent:bundleIdentifier];
+    NSError *error = nil;
+    [[NSFileManager defaultManager] createDirectoryAtURL:folder
+                             withIntermediateDirectories:true
+                                              attributes:nil
+                                                   error:&error];
+
+    // It seems safe not to handle this error because Application Support should always be
+    // available and one should always be able to create a folder in it
+    if (error != nil) {
+        DDLogError(@"Failed to create folder for %@ in Application Support: %@, %@", bundleIdentifier, error, [error userInfo]);
+        abort();
+    }
+
+    return folder;
+}
+
+// Application Support contains "the files that your app creates and manages on behalf of the user
+// and can include files that contain user data".
+//
+// See:
+// https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/MacOSXDirectories/MacOSXDirectories.html#//apple_ref/doc/uid/TP40010672-CH10-SW1
+- (NSURL *)applicationSupportURL {
+    NSError *error = nil;
+    NSFileManager *manager = [NSFileManager defaultManager];
+    NSURL *applicationSupportURL = [manager URLForDirectory:NSApplicationSupportDirectory
+                                                inDomain:NSUserDomainMask
+                                       appropriateForURL:nil
+                                                  create:false
+                                                   error:&error];
+
+    // It seems safe not to handle this error because Application Support should always be
+    // available.
+    if (error != nil) {
+        DDLogError(@"Failed to get path to Application Support folder: %@, %@", error, [error userInfo]);
+        abort();
+    }
+
+    return applicationSupportURL;
```

Comment:
You can proooobably simplify this API as follows:

```
NSURL *appSupportURL = [[[NSFileManager defaultManager] URLsForDirectory:NSApplicationSupportDirectory inDomains:NSUserDomainMask] lastObject];
```

[Ref. here](https://github.com/Simperium/simperium-ios/blob/develop/Simperium/SPPersistentMutableDictionary.m#L400)

(The Application Support path is expected to always be there!)